cmake/Modules/Dependencies.cmake

```diff
@@ -33,13 +33,22 @@ if(BUILD_DOCS)
     set(rocm_cmake_tag
       "c044bb52ba85058d28afe2313be98d9fed02e293" # develop@2023.09.12. (move to 6.0 tag when released)
       CACHE STRING "rocm-cmake tag to download")
+
+    # Define the cache directory
+    set(FETCHCONTENT_BASE_DIR "${CMAKE_BINARY_DIR}/_deps_cache")  
+
     FetchContent_Declare(
       rocm-cmake
       GIT_REPOSITORY https://github.com/RadeonOpenCompute/rocm-cmake.git
       GIT_TAG        ${rocm_cmake_tag}
       SOURCE_SUBDIR "DISABLE ADDING TO BUILD" # We don't really want to consume the build and test targets of ROCm CMake.
     )
-    FetchContent_MakeAvailable(rocm-cmake)
+    FetchContent_GetProperties(rocm-cmake)
+    if(NOT rocm-cmake_POPULATED)
+      FetchContent_Populate(rocm-cmake)
+      add_subdirectory(${rocm-cmake_SOURCE_DIR} ${rocm-cmake_BINARY_DIR} EXCLUDE_FROM_ALL)
+    endif()
+
```

Comment:
According to cmake documentation: 

The [FetchContent_MakeAvailable()](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable) command ensures the named dependencies have been populated, either by an earlier call, or by populating them itself.

So wouldn't MakeAvailable already perform the check you are doing here? Check if its already been populated.